main.go

```diff
@@ -306,11 +309,13 @@ func main() {
 		vdsOverrideOpts = controllerOptions
 	}
 	if err = (&controllers.VaultDynamicSecretReconciler{
-		Client:        mgr.GetClient(),
-		Scheme:        mgr.GetScheme(),
-		Recorder:      mgr.GetEventRecorderFor("VaultDynamicSecret"),
-		ClientFactory: clientFactory,
-		HMACValidator: hmacValidator,
+		Client:         mgr.GetClient(),
+		Scheme:         mgr.GetScheme(),
+		Recorder:       mgr.GetEventRecorderFor("VaultDynamicSecret"),
+		ClientFactory:  clientFactory,
+		HMACValidator:  hmacValidator,
+		SyncRegistry:   controllers.NewSyncRegistry(),
```

Comment:
My one minor concern while reading through the code was that it looked like we were creating some lock contention on the `ResourceReferenceCache` between the various controllers which all run in parallel, but I'm happy to see that you thought ahead and are creating a separate cache for each controller!